luxonis_train/core/core.py

```diff
@@ -419,6 +420,7 @@ def infer(
         self,
         view: Literal["train", "val", "test"] = "val",
         save_dir: str | Path | None = None,
+        img_path: Optional[str] = None
```

Comment:
Instead of `Optional[str]`, you should use `str | None` (`Optional` is deprecated in python 3.10)